src/legacy/core_plugins/interpreter/server/lib/route_expression/thread/index.js

```diff
@@ -31,15 +31,7 @@ export function getWorker() {
   if (worker) return worker;
   worker = fork(workerPath, {});
 
-  // 'exit' happens whether we kill the worker or it just dies.
-  // No need to look for 'error', our worker is intended to be long lived so it isn't running, it's an issue
-  worker.on('exit', () => {
-    // Heads up: there is no worker.off
-    worker = null;
-    // Restart immediately on exit since node takes a couple seconds to spin up
-    worker = getWorker();
-  });
-
+  // handle run requests
   worker.on('message', msg => {
     const { type, value, id } = msg;
     if (type === 'run') {
```

Comment:
Can we make a point to enumerate all of these types somewhere, and document them?  I'll open an issue to do so... I'm assuming there's an `emit` somewhere, so it would be helpful to have a single source of truth for what message types we're using and what they mean.

Reply:
Yeah, that's a good call. It would also be nice to have a single place that sets up the message listening, but `functionList` is weird because we only want to listen to it once, but it's piggybacking on the generic `message` event. A little abstraction here would definitely be nice, thanks for opening the issue.